core/io/stream_peer.cpp

```diff
@@ -492,6 +492,8 @@ Error StreamPeerBuffer::get_partial_data(uint8_t* p_buffer, int p_bytes,int &r_r
 
 	PoolVector<uint8_t>::Read r = data.read();
 	copymem(p_buffer,r.ptr(),r_received);
+	
+	// FIXME: return what? OK or ERR_*
```

Comment:
I've never used the `StreamPeerBuffer` class myself but judging from the commit log of cbbcf727035c8b481889f605337a96a9e58ed970 and it's GDScript signature it looks cool.

`get_partial_data` is inherited from `PacketPeer` and we cannot change it's type.
It also appears to be exposed to GDScript `StreamPeerBuffer.get_partial_data(...)` via `StreamPeer::_get_partial_data(...)`.
It think this should return a proper error when needed, so option 2, something like
```
diff --git a/core/io/stream_peer.cpp b/core/io/stream_peer.cpp
index a64ebf2..bf6a861 100644
--- a/core/io/stream_peer.cpp
+++ b/core/io/stream_peer.cpp
@@ -479,12 +479,15 @@ Error StreamPeerBuffer::get_data(uint8_t* p_buffer, int p_bytes){
 }
 Error StreamPeerBuffer::get_partial_data(uint8_t* p_buffer, int p_bytes,int &r_received){
 
+       Error err = OK;
 
        if (pointer+p_bytes > data.size()) {
+               err = ERR_FILE_EOF;
+
                r_received=data.size()-pointer;
                if (r_received<=0) {
                        r_received=0;
-                       return OK; //you got 0
+                       return err; //you got 0
                }
        } else {
                r_received=p_bytes;
@@ -492,6 +495,8 @@ Error StreamPeerBuffer::get_partial_data(uint8_t* p_buffer, int p_bytes,int &r_r
 
        PoolVector<uint8_t>::Read r = data.read();
        copymem(p_buffer,r.ptr(),r_received);
+
+       return err;
 }
 
 int StreamPeerBuffer::get_available_bytes() const {
```